nebula-logger/main/log-management/classes/LoggerSettingsController.cls

```diff
@@ -0,0 +1,372 @@
+                LoggingLevel__c,
+                SetupOwner.Name,
+                SetupOwner.Type,
+                SetupOwnerId,
+                StripInaccessibleRecordFields__c
+            FROM LoggerSettings__c
+        ];
+    }
+
+    private static Map<Id, String> querySetupOwnerNames(List<Id> setupOwnerIds) {
+        Map<Id, String> setupOwnerIdToName = new Map<Id, String>();
+        for (Profile profile : [SELECT Id, Name FROM Profile WHERE Id IN :setupOwnerIds]) {
+            setupOwnerIdToName.put(profile.Id, profile.Name);
+        }
+        for (User user : [SELECT Id, Username FROM User WHERE Id IN :setupOwnerIds]) {
+            setupOwnerIdToName.put(user.Id, user.Username);
+        }
+        return setupOwnerIdToName;
+    }
+
+    /**
+     * @description Inner class for returning all custom `List<PicklistOption>` in a single Apex method call
+     */
+    public class LoggerSettingsPicklistOptions {
+        @AuraEnabled
+        public List<PicklistOption> loggingLevelOptions;
+        @AuraEnabled
+        public List<PicklistOption> saveMethodOptions;
+        @AuraEnabled
+        public List<PicklistOption> setupOwnerTypeOptions;
+        @AuraEnabled
+        public List<PicklistOption> shareAccessLevelOptions;
+    }
+
+    /**
+     * @description Inner DTO class for picklist options since Schema.PicklistEntry isn't supported for aura-enabled methods
+     */
+    public class PicklistOption {
+        @AuraEnabled
+        public String label;
+        @AuraEnabled
+        public String value;
+    }
+
+    /**
+     * @description Inner class used for sorting LoggerSettings__c, used for 3 reasons:
+     *      1. Trying to sort in SOQL on SetupOwner.Type, SetupOwner.Name results in only user-specific records being returned (no idea why - seems like a bug)
+     *      2. Records tied to profiles do not return the actual profile name in SetupOwner.Name - example: System Admin returns as 'PT1' in query results
+     *      3. Records tied to the org or profiles return unhelpful values in SetupOwner.Type - org returns '00D', profiles return '00e'
+     */
+    public class SettingsRecordResult implements Comparable {
+        @AuraEnabled
+        public String setupOwnerType;
+        @AuraEnabled
+        public String setupOwnerName;
+        @AuraEnabled
+        public String createdByUsername;
+        @AuraEnabled
+        public String lastModifiedByUsername;
+        @AuraEnabled
+        public LoggerSettings__c record;
+
+        /**
+         * @description Implements sorting logic for the `Comparable` interface
+         * @param  compareTo The object to compare to when sorting the current item
+         * @return           The sorting result
+         */
+        public Integer compareTo(Object compareTo) {
+            SettingsRecordResult that = (SettingsRecordResult) compareTo;
+            String thisCompositeName = this.setupOwnerType + '.' + this.setupOwnerName;
```

Comment:
you can encapsulate the composite name in a private method and call it in `compareTo`:

```java
private String getCompositeName() {
  return this.setupOwnerType + '.' + this.setupOwnerName;
}

// and then in compareTo:
if (this.getCompositeName() == that.getCompositeName() {
  // etc ...
}
```

Maybe it's not worth it in this exact scenario, but I thought I'd mention it in case you like that better.

Reply:
Oooh, I do like that approach, I'll add that private method - it seems like a cleaner way to organize the code.